roles/roles.go

```diff
@@ -0,0 +1,58 @@
+package roles
+
+import (
+	"slices"
+
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/services"
+)
+
+// Role is the string literal of a role.
+type Role string
```

Comment:
I think we might want to mirror the design of the `scopes` package, where `type Role string` is a fully qualified name (`service::role`)

Reply:
Judging from tests and the structure of `RegisteredRoles`, it looks like right now `Role` is the second part of a fully qualified name (only `role`)